src/components/login/Login.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField } from '@material-ui/core';
+import * as React from 'react';
+import * as rest from "../../rest/interface";
+import '../../styles.css';
+
+interface State {
+  open : boolean;
+  submitted : boolean
+  username : string
+  password : string
+}
+
+class Props {
+  public client: rest.Evergreen;
+}
+
+export class Login extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+    this.handleClickOpen = this.handleClickOpen.bind(this);
```

Comment:
This works, but it would be cleaner to avoid having to tell the function what "this" refers to by using an arrow function, which will pass along the scoping information: https://stackoverflow.com/a/16158550

Reply:
Fixed